compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/FlattenStringConcatenationLowering.kt

```diff
@@ -73,11 +73,11 @@ class FlattenStringConcatenationLowering(val context: CommonBackendContext) : Fi
             return when (expression) {
                 is IrStringConcatenation -> true
                 is IrCall -> {
-                    val dispatchReceiver = expression.dispatchReceiver
-                    dispatchReceiver != null &&
-                            dispatchReceiver.type.isString() &&
+                    val receiver = expression.dispatchReceiver ?: expression.extensionReceiver
```

Comment:
@sfs Please also check package for irCall, this code would not work correctly for next cases (please also add test for it):
```
operator fun String?.plus(p: String) {
    this
}

fun test(a: String?, b: String): String {
return a + b
}
```

Reply:
Thanks, I completely missed this!